src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -487,51 +487,216 @@ void checkConstraintLanguageVersionAndRuntimeSupportForConversion(SyntaxNode syn
+                                .CheckConstraints(new ConstraintsHelper.CheckConstraintsArgs(Compilation, Conversions, includeNullability: false, syntax.Location, diagnostics));
+                        }
+                    }
+
+                    // ReadOnlySpan<char> MemoryExtensions.AsSpan(string)
+                    if (source.Type.IsStringType())
+                    {
+                        reportUseSiteOrMissing(
+                            TryFindAsSpanCharMethod(Compilation, destination),
+                            WellKnownMemberNames.MemoryExtensionsTypeFullName,
+                            WellKnownMemberNames.AsSpanMethodName,
+                            syntax,
+                            diagnostics);
                     }
                 }
             }
+
+            static void reportUseSiteOrMissing(MethodSymbol? method, object containingType, string methodName, SyntaxNode syntax, BindingDiagnosticBag diagnostics)
+            {
+                if (method is not null)
+                {
+                    diagnostics.ReportUseSite(method, syntax);
+                }
+                else
+                {
+                    Error(diagnostics,
+                        ErrorCode.ERR_MissingPredefinedMember,
+                        syntax,
+                        containingType,
+                        methodName);
+                }
+            }
         }
 
+        // {type}.op_Implicit(T[])
         internal static MethodSymbol? TryFindImplicitOperatorFromArray(TypeSymbol type)
         {
             Debug.Assert(type.IsSpan() || type.IsReadOnlySpan());
+            Debug.Assert(type.IsDefinition);
+
+            return TryFindImplicitOperator(type, 0, static (_, parameterType) =>
+                parameterType is ArrayTypeSymbol { IsSZArray: true, ElementType: TypeParameterSymbol });
+        }
+
+        // ReadOnlySpan<T> Span<T>.op_Implicit(Span<T>)
+        internal static MethodSymbol? TryFindImplicitOperatorFromSpan(TypeSymbol spanType, TypeSymbol readonlySpanType)
+        {
+            Debug.Assert(spanType.IsSpan() && readonlySpanType.IsReadOnlySpan());
+            Debug.Assert(spanType.IsDefinition && readonlySpanType.IsDefinition);
+
+            return TryFindImplicitOperator(spanType, readonlySpanType,
+                parameterPredicate: static (_, parameterType) => parameterType.IsSpan(),
+                returnTypePredicate: static (readonlySpanType, returnType) => readonlySpanType.Equals(returnType.OriginalDefinition, TypeCompareKind.ConsiderEverything));
+        }
 
-            return TryFindSingleMember(type, WellKnownMemberNames.ImplicitConversionName,
-                static (method) => method is
+        private static MethodSymbol? TryFindImplicitOperator<TArg>(TypeSymbol type, TArg arg,
+            Func<TArg, TypeSymbol, bool> parameterPredicate,
+            Func<TArg, TypeSymbol, bool>? returnTypePredicate = null)
```

Comment:
Consider simplifying this to a single `Func<TArg, MethodSymbol>` instead, and let the caller decide whether to check the type of the parameter and/or the return type.